### PR TITLE
Add search functionality

### PR DESCRIPTION
**Pull request by AlexHolderDeveloper**
Added a "search()" function to the package, including documentation & tests.

As an IT teacher, this allows me to refer my students to this package for usage in activities like "make a React app that processes user input to return data to the page". 

The search function can also kinda search for multiple keywords, as long as they're space-separated. Not sure if that is worth moving to a separate function though. It doesn't disrupt the search functionality either way.

Also, happy to change how the function handles a "no results found" situation. Currently just throwing an error - not sure if you'd prefer that or a different way to handle errors in this package.

Happy Hacktoberfest! 🎉

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -47,4 +47,18 @@ describe('puns', function() {
       }).to.throw('is not in the');
     });
   });
+
+  describe('#search()', function() {
+    it('should return a unique list of puns that include a given word', function(){
+      let searchResults = puns.search("C#");
+      assert.deepEqual(searchResults, Array.from(new Set(searchResults)));
```
Comment (AlexLakatos) [inline test/index.js:54]: That's always going to be true, no matter what.

**Comment by AlexLakatos** [comment]
Thanks for the PR!

I've tried to improve the functionality a bit, and opened a PR against your original repo: https://github.com/AlexHolderDeveloper/computer-puns/pull/1

The PR changes the search functionality to accept a list of keywords as an array instead of space-separated values(because this way I can search for multi-word keywords), and to return an empty list instead of throwing an error (because technically not finding a keyword in a list is not an error). With that merged, I think I'm happy to merge it into the main repo, and push a new release to npm.

**Comment by AlexHolderDeveloper** [comment]
Awesome, thankyou for the edits there! All merged into my fork and ready for pulling back into your repo.

**Comment by AlexLakatos** [comment]
Great! Thanks for this functionality, I think it's really useful!

**Pull request merged**